OIIInspector/ImageManager.py

```diff
@@ -0,0 +1,151 @@
+import logging
+from retry import retry
+import subprocess
+import time
+import shlex
+from contextlib import suppress
+from OIIInspector.Exceptions import AddressAlreadyInUse, OIIInspectorError
+from OIIInspector.utils import run_cmd
+
+log = logging.getLogger(__name__)
+
+
+class ImageManager:
+    _grpc_start_port = 50051
+    _grpc_max_port_tries = 100
+    _grpc_max_tries = 5
+    _grpc_init_wait_time = 1000
+    _base_container_name = "OIIInspector_running_image"
+
+    def __init__(self):
+        self._container_platform = self._get_container_platform()
+        self._image_address = None
+        self._port = self._grpc_start_port
+        self._rpc_proc = None
+
+    def start_image(self, image_address):
+        self._image_address = image_address
+        self._remove_old_image_if_exists()
+        if self._check_if_image_already_runs() is False:
+            self._pull_image()
+            self._port, self._rpc_proc = self._serve_index_registry()
+        else:  # pragma: no cover"
+            pass
+
+    def get_local_address_of_image(self):
+        return "localhost:{port}".format(port=self._port)
+
+    def close_image_manager(self):
```

Comment:
Could you implement and use this class as a context manager? This way the cleanup will be performed even if there's an issue.
https://realpython.com/python-with-statement/#coding-class-based-context-managers